dart/lib/flat_buffers.dart

```diff
@@ -346,15 +348,14 @@ class Builder {
   /// bytes 4-7 of the file.
   Uint8List finish(int offset, [String fileIdentifier]) {
     _prepare(max(_sizeofUint32, _maxAlign), fileIdentifier == null ? 1 : 2);
-    int alignedTail = _tail + ((-_tail) % _maxAlign);
-    _setUint32AtTail(_buf, alignedTail, alignedTail - offset);
+    _setUint32AtTail(_buf, size, size - offset);
```

Comment:
Does this call `size` twice? Actually 4x with the 2 uses below.
Maybe this is normal in Dart, but I'd personally like to see `size()` to make it obvious a function is being called. And a temp variable if indeed it would be called multiple times otherwise.

Reply:
My assumption was it dart compiler would know the value doesn't change between the getter value doesn't change between the calls and could optimize it out, but maybe I'm giving it too much credit (don't really have time to microbenchmark this now).

Even though a getter here is ["idiomatic dart"](https://dart.dev/guides/language/effective-dart/design#do-use-getters-for-operations-that-conceptually-access-properties) I'm perfectly fine with changing it to a method. Just please agree with @dnfield on one solution you both would be OK with :) 

Would a `size()` method work for you both? With its result cached in the `finish()` function to avoid multiple calls.

Reply:
As it happens, I disagree with idiomatic dart on this point - getters should be cheap.

I think we should either change it to a method or cache the property and keep it as a getter. I am ambivalent about which path is chosen.

Reply:
In either case, we should document the behavior so consumers know what to expect without reading the code.

Reply:
OK, changed according to that.